### DISCUSSION TITLE
Reference-able inline text spans?

### DISCUSSION THREAD

**Discussion by dan-zheng** [Textualize/textual#4089] [Q&A] [answered]
It is possible to write reference-able inline text spans / Markup links?

I would like to have labeled inline text spans, like this in HTML:
```html
<span class='x'>x</span> = 1
<span class='z'>z</span> = <span class='x'>x</span> + <span class='y'>y</span>
```
With behavior like `highlight_all('x')` when mouse hovers over one of the `x` spans.

---

Related resources:
- [Actions embedded as links](https://textual.textualize.io/guide/actions/#links): "Actions may be embedded as links within console markup. You can create such links with a `@click` tag." These markup links are inline text – but they have no id / class, and thus can't be referenced.
- [Enter](https://textual.textualize.io/events/enter/) and [Leave](https://textual.textualize.io/events/leave/) events: for implementing "highlight on hover" behavior. But this would require reference-able widgets, formatted like inline text for my use case.

**Comment by davep** [comment, marked as answer]
To the best of my knowledge there isn't a supported method of doing that. At the moment you *could* exploit a known bug where all links with the exact same action get the same styling when one of them receives hover, you can see this in action with this code:

```python
from random import choice
from string import ascii_uppercase

from textual.app import App, ComposeResult
from textual.widgets import Static

class HoverLettersApp(App[None]):

    CSS = """
    Static {
        link-style: none;
        link-color-hover: white;
        link-background-hover: red;
    }
    """

    def compose(self) -> ComposeResult:
        text = ""
        for _ in range(5000):
            letter = choice(ascii_uppercase)
            text += f"[@click=does_nothing('{letter}')]{letter}[/]"
        yield Static(text)

    def action_does_nothing(self) -> None:
        pass


if __name__ == "__main__":
    HoverLettersApp().run()
```

but this is seen as a bug (#1587) and will likely receive a fix one day.

**Reply by dan-zheng** [reply]
Interesting, thanks for sharing!

Yes, a supported method would great. Ideally, reference-able inline text would be able to support actions like "jump to definition ('first occurrence')" too, which I believe isn't supported by the above.

**Comment by dan-zheng** [comment]
Question to maintainers: given that markup links exist, how feasible would it be to make them reference-able (by an id or class)?

Markup links fulfill many of the requirements in my question:
- [x] Inline text
- [x] Can have on-click behavior
- [x] Can have on-hover styling
- [ ] Are reference-able

So making them reference-able seems like one solution approach.

**Reply by TomJGooding** [reply]
I'm _not_ a maintainer, but I think perhaps there's some confusion here between Textual widgets and [Rich renderables](https://rich.readthedocs.io/en/latest/protocol.html).

The "inline text" you see in Textual is implemented using Rich renderables, which simply have no concept of a class or id.

The markup links (I think?) work based on a concept of `meta` data from Rich renderables, which I suppose would make them reference-able? Unfortunately this doesn't seem to be well documented...
